driver.go

```diff
@@ -60,6 +61,14 @@ func (d MySQLDriver) Open(dsn string) (driver.Conn, error) {
 	mc.parseTime = mc.cfg.ParseTime
 	mc.strict = mc.cfg.Strict
 
+	// Call startWatcher for context support (From Go 1.8)
+	type starter interface {
```

Comment:
Maybe move this declaration out of function scope?
What about name `watcher` instead of `starter` ?
``` go
type watcher interface {
    startWatch()
}
```